[Build Error due to py_limited_api True]  error: identifier "Py_buffer" is undefined/

got 100+ error while py_limited_api=True with v2.8 nighlty wheels
build pass when manually set py_limited_api=False even when PyTorch version > 2.6

Am I miss something so I encounter those errors?
Is there any other configuration to solve this rather than manually set py_limited_api=True?


```
/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/buffer_info.h(107): error: identifier "Py_buffer" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/buffer_info.h(153): error: identifier "Py_buffer" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/buffer_info.h(154): error: identifier "Py_buffer" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/buffer_info.h(181): error: identifier "Py_buffer" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/buffer_info.h(118): error: more than one user-defined conversion from "<error-type>" to "std::vector<int64_t, std::allocator<int64_t>>" applies:

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/buffer_info.h(108): error: no instance of constructor "pybind11::buffer_info::buffer_info" matches the argument list
            argument types are: (<error-type>, <error-type>, <error-type>, <error-type>, {...}, <error-type>, <error-type>)

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/buffer_info.h(140): error: no instance of overloaded function "std::swap" matches the argument list
            argument types are: (<error-type> *, <error-type> *)

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/buffer_info.h(148): error: identifier "PyBuffer_Release" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(263): error: identifier "PyGILState_Check" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(278): error: identifier "PyGILState_Check" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(322): error: pointer to incomplete class type "_typeobject" is not allowed

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(326): error: pointer to incomplete class type "_typeobject" is not allowed

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(489): error: pointer to incomplete class type "_typeobject" is not allowed

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(491): error: pointer to incomplete class type "_typeobject" is not allowed

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(615): error: identifier "PyList_GET_ITEM" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(647): error: identifier "PyTracebackObject" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(666): error: pointer to incomplete class type "PyCodeObject" is not allowed

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(666): error: type name is not allowed

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(666): error: expected an expression

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(670): error: pointer to incomplete class type "PyCodeObject" is not allowed

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(670): error: type name is not allowed

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(670): error: expected an expression

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(674): error: identifier "PyFrame_GetBack" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(674): error: cannot deduce "auto" type

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(947): error: identifier "PyInstanceMethod_Check" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(948): error: identifier "PyInstanceMethod_GET_FUNCTION" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(949): error: identifier "PyMethod_Check" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(950): error: identifier "PyMethod_GET_FUNCTION" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(1261): error: identifier "PyListObject" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(1261): error: expected an expression

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(1261): error: identifier "PyTupleObject" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(1261): error: expected an expression

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(1359): error: identifier "PyStaticMethod_Type" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(1777): error: identifier "PyByteArray_AS_STRING" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(1778): error: identifier "PyByteArray_GET_SIZE" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(2098): error: identifier "PyTuple_GET_SIZE" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(2187): error: identifier "PyList_GET_SIZE" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(2269): error: identifier "PyStaticMethod_New" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(2269): error: identifier "PyStaticMethod_New" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(2274): error: identifier "PyObject_CheckBuffer" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(2274): error: identifier "PyObject_CheckBuffer" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(2277): error: identifier "PyBUF_STRIDES" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(2277): error: identifier "PyBUF_FORMAT" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(2279): error: identifier "PyBUF_WRITABLE" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(2281): error: expected a type specifier

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(2281): error: cannot deduce "auto" type

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(2282): error: identifier "PyObject_GetBuffer" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(2309): error: identifier "PyMemoryView_FromBuffer" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(2390): error: identifier "PyBUF_READ" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(2390): error: identifier "PyBUF_WRITE" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(2423): error: identifier "Py_buffer" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(2435): error: identifier "PyMemoryView_FromBuffer" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/pytypes.h(2459): error: identifier "PyObject_LengthHint" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/detail/internals.h(557): error: pointer to incomplete class type "_ts" is not allowed

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/detail/cpp_conduit.h(24): error: pointer to incomplete class type "_typeobject" is not allowed

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/detail/cpp_conduit.h(29): error: identifier "_PyType_Lookup" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/detail/cpp_conduit.h(30): error: identifier "PyInstanceMethod_Check" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/detail/type_caster_base.h(117): error: pointer to incomplete class type "_typeobject" is not allowed

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/detail/type_caster_base.h(151): error: pointer to incomplete class type "_typeobject" is not allowed

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/detail/type_caster_base.h(161): error: pointer to incomplete class type "_typeobject" is not allowed

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/detail/type_caster_base.h(465): error: identifier "_PyThreadState_UncheckedGet" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/cast.h(346): error: pointer to incomplete class type "_typeobject" is not allowed

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/cast.h(346): error: cannot deduce "auto" type

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/cast.h(368): error: pointer to incomplete class type "_typeobject" is not allowed

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/cast.h(411): error: identifier "PyUnicode_AsUTF8AndSize" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/cast.h(410): error: cannot deduce "auto" type

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/cast.h(1819): error: identifier "PyGILState_Check" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/attr.h(105): error: identifier "PyHeapTypeObject" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/attr.h(109): error: incomplete type is not allowed

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/attr.h(107): error: incomplete type is not allowed

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/attr.h(314): error: incomplete type is not allowed

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/attr.h(357): error: pointer to incomplete class type "_typeobject" is not allowed

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/attr.h(357): error: identifier "Py_TPFLAGS_MANAGED_DICT" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/detail/class.h(32): error: pointer to incomplete class type "_typeobject" is not allowed

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/detail/class.h(51): error: incomplete class type "_typeobject" is not allowed

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/detail/class.h(57): error: incomplete class type "_typeobject" is not allowed

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/detail/class.h(61): error: incomplete type is not allowed

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/detail/class.h(61): error: identifier "PyHeapTypeObject" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/detail/class.h(61): error: identifier "heap_type" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/detail/class.h(74): error: expected an expression

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/detail/class.h(74): error: cannot deduce "auto" type

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/detail/class.h(74): error: expected a ";"

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/detail/class.h(84): error: cannot deduce "auto" type

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/detail/class.h(141): error: identifier "_PyType_Lookup" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/detail/class.h(154): error: pointer to incomplete class type "_typeobject" is not allowed

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/detail/class.h(165): error: incomplete class type "_typeobject" is not allowed

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/detail/class.h(176): error: identifier "_PyType_Lookup" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/detail/class.h(177): error: identifier "PyInstanceMethod_Check" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/detail/class.h(181): error: incomplete class type "_typeobject" is not allowed

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/detail/class.h(188): error: incomplete class type "_typeobject" is not allowed

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/detail/class.h(245): error: incomplete class type "_typeobject" is not allowed

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/detail/class.h(259): error: expected an expression

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/detail/class.h(259): error: cannot deduce "auto" type

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/detail/class.h(259): error: expected a ";"

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/detail/class.h(269): error: cannot deduce "auto" type

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/detail/class.h(299): error: pointer to incomplete class type "_typeobject" is not allowed

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/detail/class.h(359): error: pointer to incomplete class type "_typeobject" is not allowed

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/detail/class.h(445): error: identifier "_PyObject_GetDictPtr" is undefined

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/detail/class.h(469): error: pointer to incomplete class type "_typeobject" is not allowed

/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/pybind11/detail/class.h(499): error: expected an expression

Error limit reached.
100 errors detected in the compilation of "/home/zhaoqion/extension-cpp/extension_cpp/csrc/cuda/muladd.cu".
Compilation terminated.
[2/2] c++ -MMD -MF /home/zhaoqion/extension-cpp/build/temp.linux-x86_64-cpython-311/extension_cpp/csrc/muladd.o.d -pthread -B /home/zhaoqion/miniforge3/envs/cpp_extension/compiler_compat -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /home/zhaoqion/miniforge3/envs/cpp_extension/include -fPIC -O2 -isystem /home/zhaoqion/miniforge3/envs/cpp_extension/include -fPIC -I/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include -I/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/include/torch/csrc/api/include -I/home/zhaoqion/miniforge3/envs/cpp_extension/include/python3.11 -c -c /home/zhaoqion/extension-cpp/extension_cpp/csrc/muladd.cpp -o /home/zhaoqion/extension-cpp/build/temp.linux-x86_64-cpython-311/extension_cpp/csrc/muladd.o -O3 -fdiagnostics-color=always -DPy_LIMITED_API=0x03090000 -DTORCH_API_INCLUDE_EXTENSION_H -DPy_LIMITED_API=0x03090000 -DTORCH_EXTENSION_NAME=_C -std=c++17
ninja: build stopped: subcommand failed.
Traceback (most recent call last):
  File "/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/utils/cpp_extension.py", line 2557, in _run_ninja_build
    subprocess.run(
  File "/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['ninja', '-v']' returned non-zero exit status 1.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/zhaoqion/extension-cpp/setup.py", line 75, in <module>
    setup(
  File "/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/setuptools/__init__.py", line 117, in setup
    return distutils.core.setup(**attrs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 186, in setup
    return run_commands(dist)
           ^^^^^^^^^^^^^^^^^^
  File "/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 202, in run_commands
    dist.run_commands()
  File "/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 1002, in run_commands
    self.run_command(cmd)
  File "/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/setuptools/dist.py", line 1104, in run_command
    super().run_command(command)
  File "/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 1021, in run_command
    cmd_obj.run()
  File "/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/setuptools/command/install.py", line 109, in run
    self.do_egg_install()
  File "/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/setuptools/command/install.py", line 167, in do_egg_install
    self.run_command('bdist_egg')
  File "/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/setuptools/_distutils/cmd.py", line 357, in run_command
    self.distribution.run_command(command)
  File "/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/setuptools/dist.py", line 1104, in run_command
    super().run_command(command)
  File "/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 1021, in run_command
    cmd_obj.run()
  File "/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/setuptools/command/bdist_egg.py", line 177, in run
    cmd = self.call_command('install_lib', warn_dir=False)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/setuptools/command/bdist_egg.py", line 163, in call_command
    self.run_command(cmdname)
  File "/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/setuptools/_distutils/cmd.py", line 357, in run_command
    self.distribution.run_command(command)
  File "/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/setuptools/dist.py", line 1104, in run_command
    super().run_command(command)
  File "/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 1021, in run_command
    cmd_obj.run()
  File "/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/setuptools/command/install_lib.py", line 19, in run
    self.build()
  File "/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/setuptools/_distutils/command/install_lib.py", line 113, in build
    self.run_command('build_ext')
  File "/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/setuptools/_distutils/cmd.py", line 357, in run_command
    self.distribution.run_command(command)
  File "/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/setuptools/dist.py", line 1104, in run_command
    super().run_command(command)
  File "/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 1021, in run_command
    cmd_obj.run()
  File "/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/setuptools/command/build_ext.py", line 99, in run
    _build_ext.run(self)
  File "/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/setuptools/_distutils/command/build_ext.py", line 368, in run
    self.build_extensions()
  File "/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/utils/cpp_extension.py", line 1053, in build_extensions
    build_ext.build_extensions(self)
  File "/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/setuptools/_distutils/command/build_ext.py", line 484, in build_extensions
    self._build_extensions_serial()
  File "/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/setuptools/_distutils/command/build_ext.py", line 510, in _build_extensions_serial
    self.build_extension(ext)
  File "/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/setuptools/command/build_ext.py", line 264, in build_extension
    _build_ext.build_extension(self, ext)
  File "/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/setuptools/_distutils/command/build_ext.py", line 565, in build_extension
    objects = self.compiler.compile(
              ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/utils/cpp_extension.py", line 837, in unix_wrap_ninja_compile
    _write_ninja_file_and_compile_objects(
  File "/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/utils/cpp_extension.py", line 2210, in _write_ninja_file_and_compile_objects
    _run_ninja_build(
  File "/home/zhaoqion/miniforge3/envs/cpp_extension/lib/python3.11/site-packages/torch/utils/cpp_extension.py", line 2574, in _run_ninja_build
    raise RuntimeError(message) from e
RuntimeError: Error compiling objects for extension
```

--- a/extension_cpp/csrc/cuda/muladd.cu
+++ b/extension_cpp/csrc/cuda/muladd.cu
@@ -1,4 +1,6 @@
-#include <torch/extension.h>
+#include <ATen/Operators.h>
+#include <torch/all.h>
+#include <torch/library.h>
 
 #include <cuda.h>
 #include <cuda_runtime.h>
@@ -18,7 +20,7 @@ at::Tensor mymuladd_cuda(const at::Tensor& a, const at::Tensor& b, double c) {
   TORCH_INTERNAL_ASSERT(b.device().type() == at::DeviceType::CUDA);
   at::Tensor a_contig = a.contiguous();
   at::Tensor b_contig = b.contiguous();
-  at::Tensor result = torch::empty(a_contig.sizes(), a_contig.options());
+  at::Tensor result = at::empty(a_contig.sizes(), a_contig.options());
   const float* a_ptr = a_contig.data_ptr<float>();
   const float* b_ptr = b_contig.data_ptr<float>();
   float* result_ptr = result.data_ptr<float>();
@@ -41,7 +43,7 @@ at::Tensor mymul_cuda(const at::Tensor& a, const at::Tensor& b) {
   TORCH_INTERNAL_ASSERT(b.device().type() == at::DeviceType::CUDA);
   at::Tensor a_contig = a.contiguous();
   at::Tensor b_contig = b.contiguous();
-  at::Tensor result = torch::empty(a_contig.sizes(), a_contig.options());
+  at::Tensor result = at::empty(a_contig.sizes(), a_contig.options());
   const float* a_ptr = a_contig.data_ptr<float>();
   const float* b_ptr = b_contig.data_ptr<float>();
   float* result_ptr = result.data_ptr<float>();

